logstash-core/lib/logstash/util/substitution_variables.rb

```diff
@@ -50,6 +50,20 @@ def replace_placeholders(value)
         raise LogStash::ConfigurationError, "Cannot evaluate `#{placeholder}`. Replacement variable `#{name}` is not defined in a Logstash secret store " +
             "or as an Environment entry and there is no default value given."
       end
+
+      if [ "hosts", :uri ].include?(name_or_validator.join)
```

Comment:
Here, it would be great to add also `:array`

Reply:
Thanks @fbaligand !  I've added it, let me know if this isn't what you meant.

Reply:
This is exactly what I meant :)
=> Nice!